Nightly CI unit test failure due to formula ambiguity

PR #1902 broke `io.deephaven.engine.table.impl.QueryTableTest#testComparableRangeFilterSimple`:
```
Error compiling class io.deephaven.temp.cm102055576987228486v55_0.Formula:
/io/deephaven/temp/cm102055576987228486v55_0/Formula.java:160: error: reference to eq is ambiguous
            return eq(LV, null) ? null : java.math.BigInteger.valueOf(LV);
                   ^
  both method eq(long,BigInteger) in QueryLanguageFunctionUtils and method eq(long,BigDecimal) in QueryLanguageFunctionUtils match
Note: /io/deephaven/temp/cm102055576987228486v55_0/Formula.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error

java.lang.RuntimeException: Error compiling class io.deephaven.temp.cm102055576987228486v55_0.Formula:
/io/deephaven/temp/cm102055576987228486v55_0/Formula.java:160: error: reference to eq is ambiguous
            return eq(LV, null) ? null : java.math.BigInteger.valueOf(LV);
                   ^
  both method eq(long,BigInteger) in QueryLanguageFunctionUtils and method eq(long,BigDecimal) in QueryLanguageFunctionUtils match
Note: /io/deephaven/temp/cm102055576987228486v55_0/Formula.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error

	at io.deephaven.compilertools.CompilerTools.maybeCreateClassHelper(CompilerTools.java:790)
	at io.deephaven.compilertools.CompilerTools.maybeCreateClass(CompilerTools.java:753)
	at io.deephaven.compilertools.CompilerTools.compileHelper(CompilerTools.java:511)
	at io.deephaven.compilertools.CompilerTools.compile(CompilerTools.java:470)
	at io.deephaven.compilertools.CompilerTools.compile(CompilerTools.java:425)
	at io.deephaven.engine.table.impl.select.DhFormulaColumn.lambda$compileFormula$19(DhFormulaColumn.java:769)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at io.deephaven.engine.table.impl.select.DhFormulaColumn.compileFormula(DhFormulaColumn.java:768)
	at io.deephaven.engine.table.impl.select.DhFormulaColumn.createFormulaFactory(DhFormulaColumn.java:731)
	at io.deephaven.engine.table.impl.select.AbstractFormulaColumn.getFormula(AbstractFormulaColumn.java:230)
	at io.deephaven.engine.table.impl.select.AbstractFormulaColumn.getViewColumnSource(AbstractFormulaColumn.java:217)
	at io.deephaven.engine.table.impl.select.AbstractFormulaColumn.getDataView(AbstractFormulaColumn.java:176)
	at io.deephaven.engine.table.impl.select.SwitchColumn.getDataView(SwitchColumn.java:93)
	at io.deephaven.engine.table.impl.select.analyzers.SelectColumnLayer.<init>(SelectColumnLayer.java:59)
	at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.createLayerForSelect(SelectAndViewAnalyzer.java:196)
	at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:109)
	at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:41)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$36(QueryTable.java:1377)
	at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
	at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$37(QueryTable.java:1362)
	at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3152)
	at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1361)
	at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1341)
	at io.deephaven.engine.table.impl.TableWithDefaults.update(TableWithDefaults.java:287)
	at io.deephaven.engine.table.impl.QueryTableTest.testComparableRangeFilterSimple(QueryTableTest.java:812)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:255)
	at junit.framework.TestSuite.run(TestSuite.java:250)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:84)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java
+++ b/engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java
@@ -1028,32 +1028,20 @@ public final class QueryLanguageParser extends GenericVisitorAdapter<Class<?>, Q
         Class<?> rhType = getTypeWithCaching(rightExpr);
 
         if ((lhType == String.class || rhType == String.class) && op == BinaryExpr.Operator.PLUS) {
-
             if (printer.hasStringBuilder()) {
                 leftExpr.accept(this, printer);
-            }
-
-            printer.append(getOperatorSymbol(op));
-
-            if (printer.hasStringBuilder()) {
+                printer.append(getOperatorSymbol(op));
                 rightExpr.accept(this, printer);
             }
-
             return String.class;
         }
 
         if (op == BinaryExpr.Operator.OR || op == BinaryExpr.Operator.AND) {
-
             if (printer.hasStringBuilder()) {
                 leftExpr.accept(this, printer);
-            }
-
-            printer.append(getOperatorSymbol(op));
-
-            if (printer.hasStringBuilder()) {
+                printer.append(getOperatorSymbol(op));
                 rightExpr.accept(this, printer);
             }
-
             return boolean.class;
         }
 

--- a/engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java
+++ b/engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java
@@ -1021,20 +1021,22 @@ public final class QueryLanguageParser extends GenericVisitorAdapter<Class<?>, Q
     @Override
     public Class<?> visit(BinaryExpr n, VisitArgs printer) {
         BinaryExpr.Operator op = n.getOperator();
+        final Expression leftExpr = n.getLeft();
+        final Expression rightExpr = n.getRight();
 
-        Class<?> lhType = getTypeWithCaching(n.getLeft());
-        Class<?> rhType = getTypeWithCaching(n.getRight());
+        Class<?> lhType = getTypeWithCaching(leftExpr);
+        Class<?> rhType = getTypeWithCaching(rightExpr);
 
         if ((lhType == String.class || rhType == String.class) && op == BinaryExpr.Operator.PLUS) {
 
             if (printer.hasStringBuilder()) {
-                n.getLeft().accept(this, printer);
+                leftExpr.accept(this, printer);
             }
 
             printer.append(getOperatorSymbol(op));
 
             if (printer.hasStringBuilder()) {
-                n.getRight().accept(this, printer);
+                rightExpr.accept(this, printer);
             }
 
             return String.class;
@@ -1043,13 +1045,13 @@ public final class QueryLanguageParser extends GenericVisitorAdapter<Class<?>, Q
         if (op == BinaryExpr.Operator.OR || op == BinaryExpr.Operator.AND) {
 
             if (printer.hasStringBuilder()) {
-                n.getLeft().accept(this, printer);
+                leftExpr.accept(this, printer);
             }
 
             printer.append(getOperatorSymbol(op));
 
             if (printer.hasStringBuilder()) {
-                n.getRight().accept(this, printer);
+                rightExpr.accept(this, printer);
             }
 
             return boolean.class;
@@ -1061,6 +1063,25 @@ public final class QueryLanguageParser extends GenericVisitorAdapter<Class<?>, Q
         }
 
         boolean isArray = lhType.isArray() || rhType.isArray() || isTypedVector(lhType) || isTypedVector(rhType);
+
+        if (op == BinaryExpr.Operator.EQUALS) {
+            if (leftExpr.isNullLiteralExpr()) {
+                if (printer.hasStringBuilder()) {
+                    printer.append("isNull(");
+                    rightExpr.accept(this, printer);
+                    printer.append(")");
+                }
+                return boolean.class;
+            }
+            if (rightExpr.isNullLiteralExpr()) {
+                if (printer.hasStringBuilder()) {
+                    printer.append("isNull(");
+                    leftExpr.accept(this, printer);
+                    printer.append(")");
+                }
+                return boolean.class;
+            }
+        }
 
         String methodName = getOperatorName(op) + (isArray ? "Array" : "");
 

--- a/engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageParser.java
+++ b/engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageParser.java
@@ -930,7 +930,7 @@ public class TestQueryLanguageParser extends BaseArrayTestCase {
         check(expression, resultExpression, Boolean.class, new String[] {"myBoolean"});
 
         expression = "(String)myString==null";
-        resultExpression = "eq((String)myString, null)";
+        resultExpression = "isNull((String)myString)";
         check(expression, resultExpression, boolean.class, new String[] {"myString"});
 
         expression = "1==1 ? myString : null";
@@ -1855,7 +1855,7 @@ public class TestQueryLanguageParser extends BaseArrayTestCase {
                 new QueryLanguageParser(
                         "io.deephaven.engine.table.impl.lang.TestQueryLanguageParser.InnerEnum.YEAH!=null", null,
                         null, staticImports, null, null).getResult();
-        assertEquals("!eq(io.deephaven.engine.table.impl.lang.TestQueryLanguageParser.InnerEnum.YEAH, null)",
+        assertEquals("!isNull(io.deephaven.engine.table.impl.lang.TestQueryLanguageParser.InnerEnum.YEAH)",
                 result.getConvertedExpression());
     }
 

